### ISSUE TITLE
Failure when extracting the `session` unit

### ISSUE THREAD

**Issue by skgrange** [grimbough/FITfileR#13]
Hello,
A couple of weeks ago after the latest package update, I am finding I am unable to extract the `session` unit from my files. It looks like there is a data type issue somewhere. Here is a reproducible example: 

```
# An example file which will be trashed in a week or two
file_remote <- "http://skgrange.github.io/www/data/6348170562_ACTIVITY.fit"
file_local <- file.path(tempdir(), basename(file_remote))
download.file(file_remote, file_local, quiet = TRUE)

# Read the file
list_activity <- FITfileR::readFitFile(file_local)

# What units are in the file? 
FITfileR::listMessageTypes(list_activity)
#>  [1] "file_id"         "device_settings" "user_profile"    "zones_target"   
#>  [5] "sport"           "session"         "lap"             "record"         
#>  [9] "event"           "device_info"     "activity"        "file_creator"   
#> [13] "gps_metadata"

# Try to extract the session unit, this fails
FITfileR::getMessagesByType(list_activity, "session")
#> Error: Problem with `mutate()` input `..1`.
#> x non-numeric argument to binary operator
#> ℹ Input `..1` is `(function (.cols = everything(), .fns = NULL, ..., .names = NULL) ...`.

# What version am I running? 
packageDescription("FITfileR")$Version
#> [1] "0.1.1"
```

Many thanks for the help! 
Stuart.

**Comment by CraigMohn** [comment]
I also see this problem after the update. Going back to the Jan 19 version removes the error. 

If it helps, I was trying to track down why session$left_right_balance is returned as a chr NA, as it is returned as a correctly valued numeric from the python package fitdecode, with a cycling fit file generated by a Garmin 820. I believe that message_index may also have this issue.

**Comment by grimbough** [comment]
Thanks for the reports.  I remember seeing the `Input `..1` is `(function (.cols = everything()` error at some point recently, but thought I'd fixed it.  Perhaps I've missed a commit somewhere.  I'll take a look when I get a chance.

**Comment by grimbough** [comment]
@skgrange this should now be fixed in 0.1.2.  I've added some tests that should catch if I accidentally introduce this again.

```r
library(FITfileR)
# An example file which will be trashed in a week or two
file_remote <- "http://skgrange.github.io/www/data/6348170562_ACTIVITY.fit"
file_local <- file.path(tempdir(), basename(file_remote))
download.file(file_remote, file_local, quiet = TRUE)

# Read the file
list_activity <- FITfileR::readFitFile(file_local)

# Try to extract the session unit, this fails
FITfileR::getMessagesByType(list_activity, "session")
#> # A tibble: 1 x 68
#>   timestamp           start_time          start_position_… start_position_…
#>   <dttm>              <dttm>                         <dbl>            <dbl>
#> 1 2021-02-28 11:01:19 2021-02-28 10:24:21             47.4             8.63
#> # … with 64 more variables: total_elapsed_time <dbl>, total_timer_time <dbl>,
#> #   total_distance <dbl>, total_cycles <chr>, nec_lat <dbl>, nec_long <dbl>,
#> #   swc_lat <dbl>, swc_long <dbl>, avg_stroke_count <dbl>, total_work <chr>,
#> #   time_standing <dbl>, enhanced_avg_speed <dbl>, enhanced_max_speed <dbl>,
#> #   message_index <chr>, total_calories <chr>, avg_speed <dbl>,
#> #   max_speed <dbl>, avg_power <chr>, max_power <chr>, total_ascent <chr>,
#> #   total_descent <chr>, first_lap_index <chr>, num_laps <chr>,
#> #   normalized_power <chr>, training_stress_score <dbl>,
#> #   intensity_factor <dbl>, left_right_balance <chr>,
#> #   avg_stroke_distance <dbl>, pool_length <dbl>, threshold_power <chr>,
#> #   num_active_lengths <chr>, avg_vertical_oscillation <dbl>,
#> #   avg_stance_time_percent <dbl>, avg_stance_time <dbl>, stand_count <chr>,
#> #   avg_vertical_ratio <dbl>, avg_stance_time_balance <dbl>,
#> #   avg_step_length <dbl>, event <chr>, event_type <chr>, sport <chr>,
#> #   sub_sport <chr>, avg_heart_rate <chr>, max_heart_rate <chr>,
#> #   avg_cadence <chr>, max_cadence <chr>, total_training_effect <dbl>,
#> #   event_group <chr>, trigger <chr>, swim_stroke <chr>,
#> #   pool_length_unit <chr>, avg_temperature <chr>, max_temperature <chr>,
#> #   avg_fractional_cadence <dbl>, max_fractional_cadence <dbl>,
#> #   total_fractional_cycles <dbl>, avg_left_torque_effectiveness <dbl>,
#> #   avg_right_torque_effectiveness <dbl>, avg_left_pedal_smoothness <dbl>,
#> #   avg_right_pedal_smoothness <dbl>, avg_combined_pedal_smoothness <dbl>,
#> #   avg_left_pco <chr>, avg_right_pco <chr>,
#> #   total_anaerobic_training_effect <dbl>

# What version am I running? 
packageDescription("FITfileR")$Version
#> [1] "0.1.2"
```

@CraigMohn I think the left right balance is an unrelated issue.  I've created a new issue (#14) where I'll report any progress.

**Comment by skgrange** [comment]
Many thanks for taking a look and fixing the issue. Enjoy the day!
